server/game/Clocks/ChessClock.js

```diff
@@ -2,27 +2,68 @@ const Clock = require('./Clock');
 
 class ChessClock extends Clock {
     constructor(player, time) {
-        super(player, time);
+        super(player, time, 5);
         this.mode = 'stop';
         this.name = 'Chess Clock';
     }
 
+    pause() {
+        this.stop();
+    }
+
+    restart() {
+        this.start();
+    }
+
+    reset() {
+        this.stop();
+    }
+
     start() {
-        this.mode = 'down';
-        super.start();
+        if(!this.manuallyPaused) {
+            if(this.mode !== 'down') {
+                this.mode = 'down';
+                super.start();
+            }
+        }
```

Comment:
The way the clocks work is that the client is in charge of the count down, not the game state.  So the client will continue to count down regardless of what happens in the back end, and then jump around when you resync.

What clock start does is grabs the current timestamp and flags it as the "start time".  When stop is then called, the timer remaining decrements by the elapsed time and a game state sync is sent to the client.

This bit here makes sure that we don't reset the "start time" if we issue multiple start() calls in a row.  It also makes sure we can't run the clocks if we've manually paused it.